hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -211,6 +214,23 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     }
   }
 
+  @Override
+  public org.apache.iceberg.Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
```

Comment:
Could we put this in BaseMetastoreCatalog? It doesn't look like there is anything Hive-specific here, so other catalog implementations could potentially benefit.

Reply:
This is a good point. Potentially, other catalogs can benefit from this. However, `FileIO` is initialized with the catalog and there maybe custom implementations passed as catalog properties. I'm not sure yet on how to move this logic to `BaseMetastoreCatalog`. Do you mind if I do that as a seperate enhancement PR to this one?

Reply:
Sure, thanks